Electron app enhancements

Shubhdeep12 opened the pull request:
<!-- 
Tick these boxes IF they're applicable for your PR.
- Changesets are only required for PRs to Spotlight lbirary packages (e.g. @spotlightjs/overlay). Not for the website/docs or demo app contributions.
- Typo correction or small bugfix PRs don't require an issue. If you're making a bigger change, please open an issue first. 
-->
Before opening this PR:
* [x] I added a [Changeset Entry](https://spotlightjs.com/contribute/changesets/) with `pnpm changeset:add`
* [x] I referenced issues that this PR addresses
#302 
#304

HazAT commented:
Pretty cool, but I added the error dialog already, it should be in the code https://github.com/getsentry/spotlight/blob/main/packages/electron/src/electron/main/index.ts#L224

Shubhdeep12 commented:
> Pretty cool, but I added the error dialog already, it should be in the code https://github.com/getsentry/spotlight/blob/main/packages/electron/src/electron/main/index.ts#L224
> 
> Forgot to check/close the box

yeah, I missed it. it seems like we will be asking the user for the first time until sentry.('sentry-enabled') is undefined.
i removed the processes to capture exceptions and added the function to show error page whenever Sentry is about to send data ( in askForPermissionToSendToSentry function)

cc @HazAT

HazAT commented on packages/electron/index.html:
```diff
@@ -11,9 +11,43 @@
+        <p class="spotlight-title">Spotlight</p>
+      </div>
+      <h1 class="error header">Oops! An error occurred.</h1>
+      <p class="error description">Press Ctrl/Cmd + R to reload the app.</p>
```
```suggestion
      <p class="error description">Press Cmd + R to reload the app.</p>
```

HazAT reviewed (approved):
👍
